tests/org.eclipse.jface.tests/src/org/eclipse/jface/tests/viewers/AbstractTreeViewerTest.java

```diff
@@ -164,6 +164,87 @@ public void testExpandToLevel() {
 	}
 
 	@Test
+	public void testAutoExpandTrivialPath() {
```

Comment:
It would be good to have two additional test cases:
1. A test that does not expand the root element but some child and is then asserted to expand the proper number of elements.
2. A test in which an element is expanded whose child depth is lower than the `autoExpandOnSingleChildLevels` value.